test/e2e/serial/tests/configuration.go

```diff
@@ -693,9 +697,215 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
+
+			// deleting the NRO scheduler
+			By(fmt.Sprintf("deleting the NRO Scheduler object: %s", nroSchedObj.Name))
+			err = fxt.Client.Delete(ctx, &schedPods[0])
+			Expect(err).ToNot(HaveOccurred()) // TODO pick the scheduler pod after it has been deleted to ensure it restarted properly check if CheckNROSchedulerAvailable does that
+			nrosched.CheckNROSchedulerAvailable(fxt.Client, nroSchedObj.Name)
+
+			By("modifying the topology manager policy under kubeletconfig to single-numa-node")
+			Eventually(func(g Gomega) {
+				err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(targetedKC), targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				kcObj, err = kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Here we're changing the topology manager policy to to single-numa-node
+				// after the scheduler has been deleted and restarted so now we can create a
+				// topology affinity error deployment to see if the deployment's pod will be pending or not
+				kcObj.TopologyManagerPolicy = v1beta1.SingleNumaNodeTopologyManagerPolicy
+
+				err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = fxt.Client.Update(ctx, targetedKC)
+				g.Expect(err).ToNot(HaveOccurred())
+			}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+
+			waitForMcpUpdate(fxt.Client, ctx, mcps)
+
+			By("creating a topology affinity error deployment and check if the pod status is pending")
+			deployment := createTAEDeployment(fxt, ctx, "testdp", serialconfig.Config.SchedulerName, cpuResources)
+
+			maxStep := 3
+			updatedDeployment := &appsv1.Deployment{}
+			for step := 0; step < maxStep; step++ {
+				time.Sleep(10 * time.Second)
+
+				By(fmt.Sprintf("ensuring the deployment %q keep being pending %d/%d", deployment.Name, step+1, maxStep))
+
+				err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(deployment), updatedDeployment)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(wait.IsDeploymentComplete(deployment, &updatedDeployment.Status)).To(BeFalse(), "deployment %q become ready", deployment.Name)
+			}
+
+			By("checking the deployment pod has failed scheduling and its at the pending status")
+			pods, err := podlist.With(fxt.Client).ByDeployment(ctx, *updatedDeployment)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(pods)).To(Equal(1))
+
+			pod := &pods[0]
+			Expect(pod.Status.Phase).To(Equal(corev1.PodPending))
+
+			schedulerName = nroSchedObj.Status.SchedulerName
+			Expect(schedulerName).ToNot(BeEmpty(), "cannot autodetect the TAS scheduler name from the cluster")
+
+			isFailed, err := nrosched.CheckPODSchedulingFailed(fxt.K8sClient, pod.Namespace, pod.Name, schedulerName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(isFailed).To(BeFalse(), "pod %s/%s with scheduler %s failed to schedule", pod.Namespace, pod.Name, schedulerName)
```

Comment:
CheckPODSchedulingFailed returns true if the pod indeed failed. if it returned false it either means there was an error or that the pod went running, which then the test should fail: https://github.com/openshift-kni/numaresources-operator/blob/main/test/utils/nrosched/nrosched.go#L88

Reply:
got it fixed